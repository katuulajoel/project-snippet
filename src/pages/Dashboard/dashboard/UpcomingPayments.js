import React from "react";
import { useSelector } from "react-redux";
import moment from "moment";
import SummaryPlaceholder from "../../../components/SummaryPlaceholder/SummaryPlaceholder";
import { Link } from "react-router-dom";
import { getdateFormated, getDays } from "../../../utils/dateUtil";
import Icon from "../../../components/Icon";
import { isDev } from "../../../utils/auth";
import styled from "styled-components";

const UpcomingPayments = () => {
  const {
    notifications: { invoices },
  } = useSelector(({ Dashboard }) => Dashboard);
  return (
    <div className="dashboard-card">
      <h3 className="title">Latest reports</h3>
      {invoices?.length == 0 ? (
        <SummaryPlaceholder description="No upcoming payments" />
      ) : (
        <div className="content">
          {invoices?.map((invoice) => {
            return (
              <div key={invoice.id} className="list-layout clearfix">
                <div className="icon-avatar">
                  <span>
                    <Icon name="round-payment" size="sm" />
                  </span>
                </div>
                <div className="detail">
                  <p>
                    Payment for{" "}
                    <Link
                      to={`/projects/${invoice.project.id}/pay/${
                        isDev() ? "payouts" : "payments"
                      }`}
                    >
                      {invoice.full_title}
                    </Link>
                    .
                  </p>
                  <span className="sub-text">
                    <Icon name="clock-outline" size="sm" />{" "}
                    {getdateFormated(moment(invoice.due_at))}
                    {invoice.is_overdue ? (
                      <Icon
                        name="exclamation-triangle"
                        size="sm"
                        className="text-danger"
                      />
                    ) : null}
                  </span>
                  {invoice.is_overdue && getDays(invoice.due_at).number > 0 && (
                    <Overdue>
                      Overdue by {getDays(invoice.due_at).number}{" "}
                      {getDays(invoice.due_at).name}
                    </Overdue>
                  )}
                </div>
              </div>
            );
          })}
        </div>
      )}
    </div>
  );
};

const Overdue = styled.span`
  text-align: center;
  font-weight: 600;
  color: #eb5757;
  font-size: 14px;
  line-height: 22px;
  background: rgba(235, 87, 87, 0.1);
  border-radius: 50px;
  padding: 2px 12px;
  margin-top: 10px !important;
  width: fit-content;
`;

export default UpcomingPayments;
