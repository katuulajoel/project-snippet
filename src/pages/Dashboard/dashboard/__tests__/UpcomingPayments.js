import React from "react";
import renderer from "react-test-renderer";
import UpcomingPayments from "../UpcomingPayments";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];

const mockAppStore = (state) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state);
};

const state = {
  Dashboard: {
    notifications: {
      invoices: [
        {
          due_at: "2020-09-22T23:59:59.999999",
          full_title: "Test project 2 (update): Milestone 3",
          id: 8,
          is_overdue: true,
          issued_at: "2020-09-08T18:21:05",
          title: "Milestone 3",
          project: { id: 123 },
        },
      ],
    },
    isMakingRequest: {},
  },
  Invoice: {
    search: {},
  },
};

describe("Latest reports test", () => {
  it("Should match snapshot test", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={mockAppStore(state)}>
            <UpcomingPayments />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
