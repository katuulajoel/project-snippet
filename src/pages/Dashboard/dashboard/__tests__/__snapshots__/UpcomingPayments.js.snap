// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Latest reports test Should match snapshot test 1`] = `
<div
  className="dashboard-card"
>
  <h3
    className="title"
  >
    Latest reports
  </h3>
  <div
    className="content"
  >
    <div
      className="list-layout clearfix"
    >
      <div
        className="icon-avatar"
      >
        <span>
          <i
            className="tg-ic-round-payment tunga-ic-sz-sm "
          />
        </span>
      </div>
      <div
        className="detail"
      >
        <p>
          Payment for
           
          <a
            href="/projects/123/pay/payments"
            onClick={[Function]}
          >
            Test project 2 (update): Milestone 3
          </a>
          .
        </p>
        <span
          className="sub-text"
        >
          <i
            className="tg-ic-clock-outline tunga-ic-sz-sm "
          />
           
          22nd Sep 2020
          <i
            className="tg-ic-exclamation-triangle tunga-ic-sz-sm text-danger"
          />
        </span>
        <span
          className="sc-bdnxRM EmtJF"
        >
          Overdue by 
          456
           
          days
        </span>
      </div>
    </div>
  </div>
</div>
`;
